File: task_management_system_frontend/src/models/TaskItem.ts
import type { User } from "./User";

export interface TaskItem {
    id: number;
    title: string;
    description?: string;
    status: TaskStatus //"TODO" | "IN_PROGRESS" | "DONE"
    priority:Priority // "LOW" | "MEDIUM" | "HIGH" 
  
    createdAt?: Date;   
    updatedAt?: Date;
  
    assigneeId?: number;
    assignee?: User;   
    creatorId?: number;
    creator?: User;
  }


  export type TaskStatus = "TODO" | "IN_PROGRESS" | "DONE";
  export type Priority = "LOW" | "MEDIUM" | "HIGH";

File: TaskManagementSystem/Domain/Entities/TaskItem.cs
namespace Domain.Entities
{
    public class TaskItem
    {
        public int Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string? Description { get; set; }
        public string Status { get; set; } = "TODO";
        public string Priority { get; set; } = "MEDIUM";

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        // FK for Assignee
        public int? AssigneeId { get; set; }
        public User? Assignee { get; set; }

        // FK for Creator
        public int CreatorId { get; set; }
        public User Creator { get; set; }

        public int? UpdaterId { get; set; }
        public User? Updater { get; set; }
    }
}
